### ISSUE TITLE
Proposal: KaiaCert DApp provides the PMP Pre-Test, securely stores the results on blockchain, and ensures transparency and verification for the exam registration process.

### ISSUE THREAD

**Issue by nhansunshine4** [kaiachain/kaia-dapp-mono#189]
**Description:**
- KaiaCert provides a PMP Pre-Test with 30 questions divided into 3 topics: Process, People, and Business Environment.
- Candidates must correctly answer 80% of the questions in each topic to pass the test.
- Once the test is passed, candidates can claim a certificate in the form of an NFT.
- The NFT acts as proof of passing the Pre-Test and is required for registering for the official PMP (Project Management Professional) certification exam in the U.S.
- The pass results for each individual topic are securely stored on the Kaia blockchain for transparency and verification.

**Bounty track:**
- NFT & Other.

![Screenshot 2024-10-11 214321](https://github.com/user-attachments/assets/2774b8a8-ff46-4660-b041-7e872ca97321)

![Screenshot 2024-10-11 214341](https://github.com/user-attachments/assets/c3d4d859-70d2-47f0-8d3c-47e375aac3bc)

![Screenshot 2024-10-11 214425](https://github.com/user-attachments/assets/08d90a2c-98a8-45aa-92d5-6463b92c829d)

![Screenshot 2024-10-11 214435](https://github.com/user-attachments/assets/57b7b989-1752-4be4-a166-0e0d16ee174e)

![Screenshot 2024-10-11 214447](https://github.com/user-attachments/assets/53b2b137-2f51-416d-ab80-3670ba30f4a0)

**Comment by zxstim** [comment]
Can you make the entire experience onchain? New pre-test taker will create from factory a smart contract (or a series of smart contract), then their answers are recorded into the smart contract as well (along with the question, or so and so).

**Comment by zxstim** [comment]
I think this could be an interesting POC

**Comment by nhansunshine4** [comment]
This DApp is currently being used within my own company, and it has been very well received. I am also in contact with training providers for the mandatory 35-hour certification courses as a prerequisite for taking the PMP exam, and they are quite interested. They mentioned that they are reaching out to the Project Management Institute in the US to discuss this idea further.

_Can you make the entire experience onchain? New pre-test taker will create from factory a smart contract (or a series of smart contract), then their answers are recorded into the smart contract as well (along with the question, or so and so)._

That's a great idea! I checked with the exam organizers, and they mentioned that they only need to record the results, while they want to keep the questions confidential. They aim for the test taker's results to be transparent, permanently stored, and publicly accessible for everyone to review.
However, I will definitely work with my team to integrate your idea into other tests, such as competency tests or annual assessments, where the questions do not need to be highly confidential.

I would like to submit the information about my build as follows: The product is fast because I have thoroughly tested it in real-world scenarios and made numerous adjustments.

Github:
https://github.com/nhansunshine4/KaiaCert

Build experience:
https://nhansunshine4.github.io/KaiaCert-DApp-for-PMP-PreTest/

Video:
https://www.youtube.com/watch?v=rwidTgRW8dY

**Comment by nhansunshine4** [comment]
@zxstim 
I have submitted it to Dorahack. You can check out the details through the link below. Thank you so much for your wonderful support.
https://dorahacks.io/buidl/17933

**Comment by zxstim** [comment]
Great work. Approved for `2000 KLAY` payout.

**Comment by zxstim** [comment]
Why use Unity instead of just a webapp? I feel like a webapp would work similarly.

**Comment by nhansunshine4** [comment]
Brother,
Thank you very much for your feedback. My background is in app and game development using Unity and C# (more than 5 years already). I experimented and put my best effort into developing it on the web, but it lacked professionalism and had quite a few bugs. In the end, I had to return to the area I am most familiar with, to ensure the best product quality for the community.
I am still trying to learn and improve every day in the area of TypeScript, but it seems like it will take quite some time to achieve the best quality.
I greatly appreciate your support.
Cheer.
